Hestia/Hestia/frontend/DevicesScreen/UITableViewControllerDevicesMain.cs

```diff
@@ -72,9 +74,22 @@ public override void ViewDidLoad()
                 this.setEditingState();
             });
 
+            //Pull to refresh
+            RefreshControl = new UIRefreshControl();
+            RefreshControl.ValueChanged += refreshTable;
+            TableView.Add(RefreshControl);
+
             // Set right button initially to edit 
             NavigationItem.RightBarButtonItem = edit;
          
         }
+
+        //Method Pull to refresh
+        private void refreshTable(object sender, EventArgs e)
+        {
+            RefreshControl.BeginRefreshing();
+            RefreshControl.EndRefreshing();

```

Comment:
Shouldn't the Tableview.ReloadData() be called in between Begin and EndRefreshing()? Also you should call the refreshDeviceList() method to actually get the data from the server.